app/controllers/addresses_controller.rb

```diff
@@ -18,4 +18,8 @@ def update
   def address_params
     params.require(:address).permit(:street, :zipcode, :city)
   end
+
+  def redirect_path_model
+    @address.addressable.player if @address.addressable
```

Comment:
As we are on Ruby 2.3 you could use the new Safe Navigation Operator here if you want, see: http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/